streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateUpdateTask.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Updates the state for all Global State Stores.
+ */
+public class GlobalStateUpdateTask implements GlobalStateMaintainer {
+
+    private class SourceNodeAndDeserializer {
+        private final SourceNode sourceNode;
+        private final RecordDeserializer deserializer;
+
+        SourceNodeAndDeserializer(final SourceNode sourceNode,
+                                  final RecordDeserializer deserializer) {
+            this.sourceNode = sourceNode;
+            this.deserializer = deserializer;
+        }
+    }
+
+    private final ProcessorTopology topology;
+    private final InternalProcessorContext processorContext;
+    private final Map<TopicPartition, Long> offsets = new HashMap<>();
+    private final Map<String, SourceNodeAndDeserializer> deserializers = new HashMap<>();
+    private final GlobalStateManager stateMgr;
+
+
+    public GlobalStateUpdateTask(final ProcessorTopology topology,
+                                 final InternalProcessorContext processorContext,
+                                 final GlobalStateManager stateMgr) {
+
+        this.topology = topology;
+        this.stateMgr = stateMgr;
+        this.processorContext = processorContext;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Map<TopicPartition, Long> initialize() {
+        final Set<String> storeNames = stateMgr.initialize(processorContext);
+        final Map<String, String> storeNameToTopic = topology.sourceStoreToSourceTopic();
+        for (final String storeName : storeNames) {
+            final String sourceTopic = storeNameToTopic.get(storeName);
+            final SourceNode source = topology.source(sourceTopic);
+            deserializers.put(sourceTopic, new SourceNodeAndDeserializer(source, new SourceNodeRecordDeserializer(source)));
```

Comment:
Because we need both the sourceNode and the deserializer when we receive data from the topic. i.e., we need to deserialize the data and the `SourceNode` to process it